test/testdata/rbs/signatures_type_aliases.rb

```diff
@@ -135,3 +135,19 @@ def method3(x)
     T.reveal_type(x) # error: Revealed type: `T.any(Integer, String)`
   end
 end
+
+module TypeAliasWithNamespace
+  module Foo
+    #: type a = Integer | String
+
+    #: (a) -> void
+    def foo(x)
+      T.reveal_type(x) # error: Revealed type: `T.any(Integer, String)`
+    end
```

Comment:
When working on this, I also found that namespaces with just type alias comments are not supported.

For example, `Foo::a` would still be considered not existing due to lack of nodes in `Foo`'s body.

```rb
module Foo
  #: type a = Integer | String
end
```

Considering it's a different issues that needs to be addressed in a different layer (CommentAssociator), I plan to open a separate PR for it.

cc @Morriar